src/pgo/trans/passes/type/TLAExpressionTypeConstraintVisitor.java

```diff
@@ -232,7 +232,9 @@ public PGoType visit(TLALet pGoTLALet) throws RuntimeException {
 	public PGoType visit(TLAGeneralIdentifier pGoTLAVariable) throws RuntimeException {
```

Comment:
These changes are wrong.
- [This change](https://github.com/UBC-NSS/pgo/pull/88/files#diff-d6c902ab7623965568904ae26a41dc00R235) creates spurious new type variables with no purpose.
- [This change](https://github.com/UBC-NSS/pgo/pull/88/files#diff-d6c902ab7623965568904ae26a41dc00R279) creates redundant type variables for number literals in procedure calls.

[Here](https://gist.github.com/mrordinaire/b619498dd3c02bac73d562ec776b9f73)'s the correct fix.